File: IKeep.Lib/Core/Entity.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace IKeep.Lib.Core
{
    public class Entity
    {
        public Guid Id { get; set; }
        public EntityStatus EntityStatus { get; set; } 
        public string EntityType
        {
            get
            {
                return GetType().Name;
            }
        }
    }

    public enum EntityStatus
    {
        Inactive,
        Active
    }
}

File: IKeep.Web/ClientApp/src/models/Entity.js
class Entity
{
    constructor(json)
    {
        if (json)
        {
            this.Id = json.id
            this.EntityStatus = json.entityStatus;
            this.EntityType = json.entityType;
        }
            
        else
        {
            this.Id = "00000000-0000-0000-0000-000000000000";
            this.EntityStatus = 1;
            this.EntityType = "";
        }
            
        
    }
}

export default Entity;